### ISSUE TITLE
Custom certificate bundle for storage initializer

### ISSUE THREAD

**Issue by danilopeixoto** [kserve/kserve#2897]
/kind feature

**Describe the solution you'd like**

The current version of KServe provides the `serving.kserve.io/s3-cabundle` annotation that allows administrators to specify the path to a certificate bundle for HTTPS validations during S3 resource initialization. However, I'm wondering if there is a way to mount a certificate bundle in the storage initializer container without having to rebuild the image.

Can KServe mount volume from secret into the container it generates? This would allow administrators to easily mount the necessary certificate bundle without the need to rebuild the container image.

**Anything else you would like to add:**

Example based on [Knative](https://knative.dev/docs/serving/tag-resolution/#procedure):

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: controller
  namespace: knative-serving
spec:
  template:
    spec:
      containers:
        - name: controller
          volumeMounts:
            - name: custom-certs
              mountPath: /path/to/custom/certs
          env:
            - name: SSL_CERT_DIR
              value: /path/to/custom/certs
      volumes:
        - name: custom-certs
          secret:
            secretName: custom-certs
```

**Comment by yuzisun** [comment]
@danilopeixoto this should be possible, we can probably add a configuration for the certificate secret name storage initializer can mount. Though this approach would require storing the ca bundle secret in every user namespace. The other option is to bake the ca bundle into the storage initializer container image but that also requires a custom container image build for every single kserve release.

**Comment by danilopeixoto** [comment]
@yuzisun Thanks for outlining the options and consequences. Would it be feasible to include both options for administrators?

**Example**

```go
type StorageInitializerConfig struct {
	Image                      string `json:"image"`
	CpuRequest                 string `json:"cpuRequest"`
	CpuLimit                   string `json:"cpuLimit"`
	MemoryRequest              string `json:"memoryRequest"`
	MemoryLimit                string `json:"memoryLimit"`
	StorageSpecSecretName      string `json:"storageSpecSecretName"`
	CaBundleSecretName         string `json:"caBundleSecretName"`
	CaBundleVolumeMountPath    string `json:"caBundleVolumeMountPath"`
	EnableDirectPvcVolumeMount bool   `json:"enableDirectPvcVolumeMount"`
}
```

**Comment by sivanantha321** [comment]
fixed by #3250 

/close
